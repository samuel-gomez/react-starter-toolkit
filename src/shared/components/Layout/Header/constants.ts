export const TITLE = 'Toolkit React Starter';
export const SUBTITLE = 'by Slash Design System';
