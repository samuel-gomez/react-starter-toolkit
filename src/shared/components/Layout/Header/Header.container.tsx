import { ComponentProps, useContext } from 'react';
import { UserContext } from 'providers/UserProvider';
import Header from './Header';
import { SUBTITLE, TITLE } from './constants';

type THeaderContainer = Omit<ComponentProps<typeof Header>, 'title' | 'subtitle'> & {
  UserContextObj?: typeof UserContext;
  HeaderCmpt?: typeof Header;
  title?: string;
  subtitle?: string;
};

const HeaderContainer = ({ UserContextObj = UserContext, HeaderCmpt = Header, title = TITLE, subtitle = SUBTITLE, ...props }: THeaderContainer) => {
  const userContext = useContext(UserContextObj);
  return <HeaderCmpt {...props} {...userContext} title={title} subtitle={subtitle} />;
};

export default HeaderContainer;
