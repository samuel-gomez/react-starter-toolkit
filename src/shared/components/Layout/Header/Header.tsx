import { ReactNode } from 'react';
import type { Tanomaly } from 'shared/types';
import { Header, Name, User, Infos } from '@axa-fr/react-toolkit-layout-header';
import logo from 'shared/images/slash-logo.svg';
import Skeleton from 'shared/components/Skeleton';
import Resilience from 'shared/components/Resilience';
import './Header.scss';

type THeaderInfo = {
  isLoaded?: boolean;
  children: ReactNode;
};

export const HeaderInfo = ({ isLoaded, children }: THeaderInfo) => (isLoaded ? <>{children}</> : <Skeleton classModifier="info" />);

type THeaderApp = {
  title: string;
  subtitle: string;
  infos?: {
    word: string;
    definition: string;
  }[];
  authName?: string;
  authRole?: string;
  link?: string;
  anomaly?: Tanomaly | null;
  fullScreen?: boolean;
};

export const HeaderApp = ({
  fullScreen,
  infos,
  title,
  subtitle,
  link = '#',
  authName = 'Non ConnectÃ©',
  authRole = 'Profil',
  anomaly,
}: THeaderApp) => (
  <Header classModifier={fullScreen ? 'fullscreen' : ''}>
    <Name title={title} img={logo.src} alt={title} subtitle={subtitle} classModifier="slash" />
    {infos && (
      <Resilience anomaly={anomaly} resilienceModifier="simple infos">
        <HeaderInfo isLoaded={infos.length > 0}>
          <Infos infos={infos} />
        </HeaderInfo>
      </Resilience>
    )}
    <User name={authName} href={link} profile={!!authRole ? authRole : 'inconnu'} />
  </Header>
);

export default HeaderApp;
