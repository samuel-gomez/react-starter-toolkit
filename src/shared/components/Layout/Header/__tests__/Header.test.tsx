import { render } from '@testing-library/react';
import { HeaderApp as Header, HeaderInfo } from '../Header';

const defaultProps = {
  title: 'title',
  subtitle: 'subtitle',
  infos: [],
  anomaly: null,
};
describe('<Header/>', () => {
  it('Render <Header/>', () => {
    const { asFragment } = render(<Header {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render <Header/> fullscreen', () => {
    const { asFragment } = render(<Header {...defaultProps} fullScreen />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should contain <Infos /> with "mydefinition" text when infos equal [{ word: "word", definition: "mydefinition" }]', () => {
    const { getByText } = render(<Header {...defaultProps} infos={[{ word: 'word', definition: 'mydefinition' }]} />);
    expect(getByText('mydefinition')).toBeDefined();
  });

  it('Should contain <User /> with when authName="FDS" authRole="ADMIN"', () => {
    const { getByText } = render(<Header {...defaultProps} authName="FDS" authRole="ADMIN" />);
    expect(getByText('FDS')).toBeDefined();
    expect(getByText(/ADMIN/)).toBeDefined();
  });

  it('Should contain <User /> with "Non Connecté" and "Profil" when authName and/or authRole are undefined', () => {
    const { getByText } = render(<Header {...defaultProps} />);
    expect(getByText('Non Connecté')).toBeDefined();
    expect(getByText(/Profil/)).toBeDefined();
  });
});

describe('<HeaderInfo/>', () => {
  const Custom = () => <span>Mon composant</span>;

  it('Should contain Skeleton when isLoader false', () => {
    const { getByRole } = render(
      <HeaderInfo>
        <Custom />
      </HeaderInfo>,
    );
    expect(getByRole('status')).toBeDefined();
  });

  it('Should contain Custom when isLoader true', () => {
    const { getByText } = render(
      <HeaderInfo isLoaded>
        <Custom />
      </HeaderInfo>,
    );
    expect(getByText('Mon composant')).toBeDefined();
  });
});
