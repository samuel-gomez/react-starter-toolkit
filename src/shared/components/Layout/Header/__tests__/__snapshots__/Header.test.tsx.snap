// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> Render <Header/> 1`] = `
<DocumentFragment>
  <div
    class="af-header"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name af-header__name--slash"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="title"
              class="af-logo__brand"
              src="/img.jpg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            title
            <span
              class="af-header__subtitle"
            >
              subtitle
            </span>
          </h2>
        </div>
        <div
          aria-busy="true"
          aria-label="loader"
          class="af-skeleton af-skeleton--info"
          role="status"
        />
        <div
          class="af-info-user"
        >
          <a
            class="af-info-user__link"
            href="#"
            title="Voir mon profil"
          >
            <span
              class="af-info-user__name"
            >
              Non Connecté
            </span>
            <span
              class="af-info-user__profile"
            >
              [Profil]
            </span>
          </a>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Header/> Render <Header/> fullscreen 1`] = `
<DocumentFragment>
  <div
    class="af-header af-header--fullscreen"
  >
    <div
      class="container af-header__wrapper"
    >
      <header
        class="af-header__content"
        role="banner"
      >
        <div
          class="af-header__name af-header__name--slash"
        >
          <a
            class="af-logo"
            href="/"
          >
            <img
              alt="title"
              class="af-logo__brand"
              src="/img.jpg"
            />
          </a>
          <h2
            class="af-header__title"
          >
            title
            <span
              class="af-header__subtitle"
            >
              subtitle
            </span>
          </h2>
        </div>
        <div
          aria-busy="true"
          aria-label="loader"
          class="af-skeleton af-skeleton--info"
          role="status"
        />
        <div
          class="af-info-user"
        >
          <a
            class="af-info-user__link"
            href="#"
            title="Voir mon profil"
          >
            <span
              class="af-info-user__name"
            >
              Non Connecté
            </span>
            <span
              class="af-info-user__profile"
            >
              [Profil]
            </span>
          </a>
        </div>
      </header>
    </div>
  </div>
</DocumentFragment>
`;
